x/ccv/provider/keeper/key_assignment.go

```diff
@@ -272,12 +297,17 @@ func (k Keeper) AppendConsumerAddrsToPrune(ctx sdk.Context, chainID string, vscI
 	if bz != nil {
 		err := consumerAddrsToPrune.Unmarshal(bz)
 		if err != nil {
+			// An error here would indicate something is very wrong,
+			// the data bytes are assumed to be correctly serialized by previous calls to this method.
 			panic(err)
 		}
 	}
+	// TODO: do we need to validate the consumerAddr we're appending here?
```

Comment:
`AppendConsumerAddrsToPrune` is called either in `AssignConsumerKey` on `DeliverTx` or in `InitGenesis`.

In `AssignConsumerKey`, `consumerAddr` will be correct as long as we validate all assigned `consumerKeys`. Note that the `utils.TMCryptoPublicKeyToConsAddr`. Maybe we should change this function to return an error. 
```golang
oldConsumerKey := k.GetValidatorConsumerPubKey(ctx, chainID, providerAddr)
consumerAddr := utils.TMCryptoPublicKeyToConsAddr(oldConsumerKey)
```

In `InitGenesis`, `consumerAddr` comes `genState.ConsumerAddrsToPrune` which should be validated by `GenesisState.Validate()`, but it's not.

Reply:
It seems like the pruned consumerAddr is indeed validated by all calling functions then? See https://github.com/cosmos/interchain-security/commit/ba19b5818e249fdc9f4c31a500a9625bd928859f which removes the TODO comment.

If you think it's worth changing `utils.TMCryptoPublicKeyToConsAddr` to return an error, I can do so 👍. Note [this line](https://github.com/cosmos/interchain-security/blob/ba19b5818e249fdc9f4c31a500a9625bd928859f/x/ccv/provider/keeper/grpc_query.go#L106) could panic during query for example. Hopefully the version of sdk we use allows the binaries to recover from panics during queries, but I'm not sure

I believe the last part of your comment re genesis validation should be handled by https://github.com/cosmos/interchain-security/pull/646.